src/marshmallow/fields.py

```diff
@@ -1434,17 +1434,36 @@ def _make_object_from_format(value, data_format):
 
 class TimeDelta(Field):
     """A field that (de)serializes a :class:`datetime.timedelta` object to an
-    integer and vice versa. The integer can represent the number of days,
-    seconds or microseconds.
+    integer or float and vice versa. The integer or float can represent the
+    number of days, seconds or microseconds.
 
-    :param precision: Influences how the integer is interpreted during
+    :param precision: Influences how the integer or float is interpreted during
         (de)serialization. Must be 'days', 'seconds', 'microseconds',
         'milliseconds', 'minutes', 'hours' or 'weeks'.
+    :param serde_type: Whether to (de)serialize to a `int` or `float`.
     :param kwargs: The same keyword arguments that :class:`Field` receives.
 
+    Integer Caveats
+    ---------------
+    Any fractional parts (which depends on the precision used) will be truncated
+    when serializing using `int`.
+
+    Float Caveats
+    -------------
+    Use of `float` when (de)serializing may result in data precision loss due
+    to the way machines handle floating point values.
+
+    Microseconds is the smallest unit able to be represented wholly.
+    Smaller units than microseconds cannot be properly deserialized and will result
+    in the truncation of any fractional data. This is only possible when using `float`.
+    For example, 1.12345 interpreted as microseconds will result in `timedelta(microseconds=1)`.
```

Comment:
I'm not sure about the meaning of "This is only possible when using `float`." I suppose you mean only float is affected because there is no fractional part in int. But I found the sentence confusing. Just removing it might make things clearer. The example clarifies things anyway.

Or perhaps rephrase and state that whatever the precision chosen, the fractional part when using float will always be truncated to microseconds.